test/contract/EasyAuction.spec.ts

```diff
@@ -378,6 +397,166 @@ describe("EasyAuction", async () => {
+        biddingToken.address,
+        60 * 60,
+        60 * 60,
+        ethers.utils.parseEther("1"),
+        ethers.utils.parseEther("1"),
+        1,
+        0,
+        false,
+        verifierMocked.address,
+      );
+
+      const sellAmount = ethers.utils.parseEther("1").add(1);
+      const buyAmount = ethers.utils.parseEther("1");
+      await expect(
+        easyAuction.placeSellOrders(
+          auctionId,
+          [buyAmount],
+          [sellAmount],
+          [queueStartElement],
+          "0x",
+        ),
+      ).to.be.revertedWith("user not allowed to place order");
+    });
+    it("order placement reverts, if allow manager is an EOA", async () => {
+      const {
+        auctioningToken,
+        biddingToken,
+      } = await createTokensAndMintAndApprove(
+        easyAuction,
+        [user_1, user_2],
+        hre,
+      );
+
+      const auctionId: BigNumber = await sendTxAndGetReturnValue(
+        easyAuction,
+        "initiateAuction(address,address,uint256,uint256,uint96,uint96,uint256,uint256,bool,address)",
+        auctioningToken.address,
+        biddingToken.address,
+        60 * 60,
+        60 * 60,
+        ethers.utils.parseEther("1"),
+        ethers.utils.parseEther("1"),
+        1,
+        0,
+        false,
+        user_3.address,
+      );
+
+      const sellAmount = ethers.utils.parseEther("1").add(1);
+      const buyAmount = ethers.utils.parseEther("1");
+      await expect(
+        easyAuction.placeSellOrders(
+          auctionId,
+          [buyAmount],
+          [sellAmount],
+          [queueStartElement],
+          "0x",
+        ),
+      ).to.be.revertedWith("function call to a non-contract account");
+    });
+    it("allow manager can not mutate state", async () => {
+      const StateChangingAllowListManager = await ethers.getContractFactory(
+        "StateChangingAllowListVerifier",
+      );
+
+      const stateChangingAllowListManager = await StateChangingAllowListManager.deploy();
+      const {
+        auctioningToken,
+        biddingToken,
+      } = await createTokensAndMintAndApprove(
+        easyAuction,
+        [user_1, user_2],
+        hre,
+      );
+
+      const auctionId: BigNumber = await sendTxAndGetReturnValue(
+        easyAuction,
+        "initiateAuction(address,address,uint256,uint256,uint96,uint96,uint256,uint256,bool,address)",
+        auctioningToken.address,
+        biddingToken.address,
+        60 * 60,
+        60 * 60,
+        ethers.utils.parseEther("1"),
+        ethers.utils.parseEther("1"),
+        1,
+        0,
+        false,
+        stateChangingAllowListManager.address,
+      );
+
+      const sellAmount = ethers.utils.parseEther("1").add(1);
+      const buyAmount = ethers.utils.parseEther("1");
+      await expect(
+        easyAuction.placeSellOrders(
+          auctionId,
+          [buyAmount],
+          [sellAmount],
+          [queueStartElement],
+          "0x",
+        ),
+      ).to.be.reverted;
```

Comment:
I was unable to write a better test. Please look here double cautiously :) 
https://github.com/nomiclabs/hardhat/issues/1252

Reply:
I think Hardhat adds a custom string for errors that would normally have no standard revert explanations (for example `function call to a non-contract account`). Not all failure cases have their own revert string however, and this is one of these cases (calling a function that mutates state with `delegatecall`). "Transaction reverted and Hardhat couldn't infer the reason." is, in a way, the right revert message for us because it's not the failure we would expect from a failed authentication.

As you say, however, the code is tricky as it is right now, since even if `StateChangingAllowListVerifier` hadn't changed the state the transaction would have been reverted anyway, since it doesn't return the right value (see other comment). Can you assert that static calling this function returns the right value `MAGICVALUE`?